### ISSUE TITLE
Missing ListBoxItem Pill Indicator Animation

### ISSUE THREAD

**Issue by robloo** [amwx/FluentAvalonia#264]
**Describe the bug**

The ListBoxItem pill indicator should be animated (conceptually similar to the radio button) when the ListBoxItem is clicked again after already being selected. This animation is currently missing and only fires when the item is first selected (not when re-selected).

**Screenshots**

WinUI (what is expected)

![cap](https://user-images.githubusercontent.com/17993847/209735638-f4033503-a7da-4f02-a5b5-b13a04077047.gif)

**Desktop/Platform (please complete the following information):**
 - OS: [e.g. Windows (11/10), linux (distro), osx, wasm, etc]
 - FluentAvalonia Version [e.g. 1.0.0, or latest]
 - Avalonia Version [e.g. 0.10.10]

**Additional context**

I can fix this one when direction is agreed. Not familiar with Avalonia's animation system yet though.

**Comment by amwx** [comment]
This was a decision I made - Since Avalonia selects on pointer down instead of pointer up, it results in the animation starting from the "SelectedPressed" state rather than "Unselected" upon initial selection of an item (so it animates from 0.5 to 1 instead of 0 to 1 all the time).

Once selection gets fixed to work on pointer up (which is on the to do list upstream), this can be fixed. I've already added the line in preview 5, it just needs uncommenting.
